refine script ignores compilation errors and attempts to run the app anyway

The `refine` script on Linux does not do a good job at stopping if it needs to compile OpenRefine and compilation errors occur.

### To Reproduce
Steps to reproduce the behavior (from a git clone of the repository):
1. First, clean with `./refine clean`
2. Then, introduce some compilation error somewhere, such as by doing `rm ./extensions/wikibase/src/org/openrefine/wikibase/qa/QAWarning.java`
3. Finally, run `./refine`

### Current Results

The compilation fails in the Wikibase extension but the `refine` script still attempts to start OpenRefine:

```
...
[ERROR] extensions/wikibase/src/org/openrefine/wikibase/qa/scrutinizers/NewEntityScrutinizer.java:[104,17] cannot find symbol
[ERROR]   symbol:   class QAWarning
[ERROR]   location: class org.openrefine.wikibase.qa.scrutinizers.NewEntityScrutinizer
[ERROR] extensions/wikibase/src/org/openrefine/wikibase/qa/scrutinizers/NewEntityScrutinizer.java:[104,39] cannot find symbol
[ERROR]   symbol:   class QAWarning
[ERROR]   location: class org.openrefine.wikibase.qa.scrutinizers.NewEntityScrutinizer
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :wikibase


15:54:05.840 [            refine_server] Starting Server bound to '127.0.0.1:3333' (0ms)
15:54:05.975 [            refine_server] refine.memory size: 1400M JVM Max heap: 1468006400 bytes (135ms)
15:54:06.359 [            refine_server] Initializing context: '/' from '/home/user/OpenRefine/main/webapp' (384ms)
15:54:06.948 [            refine_server] Starting autoreloading scanner...  (589ms)
15:54:13.478 [                   refine] Starting OpenRefine 3.8-SNAPSHOT [acca2dd]... (6530ms)
15:54:13.495 [                   refine] initializing FileProjectManager with dir (17ms)
...
```

### Expected Behavior

The `refine` script should stop at the end of the failing Maven build.

### Versions<!--  (please complete the following information)-->
 - Operating System: Linux
 - JRE or JDK Version: 17
 - OpenRefine: master branch

## Additional context

This could be solved with some bash flags, see https://github.com/OpenRefine/OpenRefine/issues/5241

--- a/refine
+++ b/refine
@@ -466,7 +466,7 @@ run() {
             mvn_prepare
             build_prepare
             $MVN process-resources
-            $MVN compile test-compile
+            $MVN compile test-compile || exit
             echo ""
         fi
     fi

